core/drivers/stm32_etzpc.c

```diff
@@ -294,7 +294,7 @@ static void get_hwcfg(struct etzpc_hwcfg *hwcfg)
 static void init_devive_from_hw_config(struct etzpc_instance *dev,
 					      paddr_t pbase)
 {
-	struct etzpc_hwcfg hwcfg;
+	struct etzpc_hwcfg hwcfg = { 0 };
```

Comment:
`memset()` for this one since it's a struct?

Reply:
> `memset()` for this one since it's a struct?

Actually, I'm trying to remember what we have decided for structs. At some point we said `memset()` to zero is preferable over `= { 0 };`, and I think there were two reasons:
1. No need to worry about `= { 0 };` vs. `= { NULL };` anymore. Weak argument, because GCC has `= { };`.
2. Padding is initialized by `memset()`, whereas it is left undefined by ` = { ... };` which will make a difference if structures are compared using `memcmp()`. Not a very strong argument either IMO, because we normally do not do that, and if we did, then initializing with valid values with `= { val };` would not be OK and would need to zero the memory first.

So... what should this line be? My preference goes to `struct etzpc_hwcfg hwcfg = { };` but perhaps I missed something?

Reply:
A problem with `= { };` (if I recall correctly) is that it can cause warnings with old GCC versions.

I agree that case 2 is not relevant for OP-TEE. Doing the `memset()` thing is ugly and somewhat clumsy. So I'm in favor of finding something else.

Reply:
> A problem with = { }; (if I recall correctly) is that it can cause warnings with old GCC versions.

Hopefully we don't care about these older versions?

Reply:
It seems that `= { }` isn't valid C code even if it's accepted by gcc (and clang?).
https://stackoverflow.com/questions/17589533/is-an-empty-initializer-list-valid-c-code

The universal zero initializer is `= { 0 }` in C (but `= { }` in C++). There's a gcc bug that relates to this.
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=53119

However, that bug is quite old and the warning can be silenced with with -Wno-missing-braces

Reply:
I would suggest to use `memset()` in this P-R for consistency with other core sources, and eventually move all those `memset()` to an alternate way once we end up with some agreement with how to handle all those inits.

Maybe at least all `enum` types initialization could move to the C universal `{ 0 }` . As for the structures, I really think `struct foo bar = <something>` is cleaner then the `memset()` way.

Reply:
I still think `= { };` is the most convenient. It's used all over the place in Linux.

> I would suggest to use memset() in this P-R for consistency with other core sources, [...]

The patch is about initializing variables that objectively don't need to be initialized. This falls in the "coding rules" category. If we don't agree on the rules, what sense can this patch make? ;-)

Reply:
Sorry, i got it wrong. optee_os has not yet moved to init all local variables, only optee_client did (https://github.com/OP-TEE/optee_client/pull/151) and optee_test is in progress (https://github.com/OP-TEE/optee_test/pull/339 & https://github.com/OP-TEE/optee_test/pull/340).

Reply:
OK, let's use `= { };` then.